### PR TITLE
 nuopc_updates for scripts_regression_tests, esmf on cheyenne

### PR DESCRIPTION
**Pull request by jedwards4b**
Updates the scripts_regression_tests.py to work with nuopc interface, updates the esmf build on cheyenne.

Test suite: CIME_MODEL=cesm; CIME_DRIVER=nuopc; ./scripts_regression_tests.py 
Test baseline: 
Test namelist changes: 
Test status: bit for bit,
Fixes 

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

**Comment by jedwards4b** [comment]
I implemented the increase walltime change incorrectly - undoing and pushing the correct change.

**Comment by jedwards4b** [comment]
@mvertens @jgfouca this is ready for review.

File: scripts/tests/scripts_regression_tests.py [hunk 1]
```diff
@@ -2453,7 +2456,7 @@ def test_configure(self):
     ###########################################################################
     def test_self_build_cprnc(self):
     ###########################################################################
-        testname = "ERS.f19_g16_rx1.A"
+        testname = "ERS_Ln7.f19_g16.X"
```
Comment (jgfouca) [inline scripts/tests/scripts_regression_tests.py:2459]: Why this change?

Comment (jedwards4b) [reply scripts/tests/scripts_regression_tests.py:2459]: Because the test is built and run and it's only intended to test the cprnc build so building and running the cheapest possible test makes sense.

Comment (jgfouca) [reply scripts/tests/scripts_regression_tests.py:2459]: This test doesn't pass on melvin unfortunately.

Comment (jedwards4b) [reply scripts/tests/scripts_regression_tests.py:2459]: Shouldn't it?  I can change this back to ERS_Ln7.f19_g16_rx1.A - will that work?

Comment (jedwards4b) [reply scripts/tests/scripts_regression_tests.py]: @jgfouca please try again now.

Comment (jgfouca) [reply scripts/tests/scripts_regression_tests.py]: @jedwards4b , yes it shouldn't, but currently does not. See https://github.com/ESMCI/cime/issues/3487 . I think any "A" case will work; it's the X cases that stopped working.
